### ISSUE TITLE
Add thebe support

### ISSUE THREAD

**Issue by yuvipanda** [notebook-sharing-space/nbss#9]
We can have live widgets and stuff! https://github.com/executablebooks/sphinx-thebe is an example to steal code from

**Comment by psychemedia** [comment]
For many users, they may not know what they need in a container to run a notebook and may not have a Binder repo to point to.

It might be useful to provide a list a of possible default Binder environments that will catch the majority of examples, eg a selection of the https://github.com/binder-examples environments. (I also started to explore some "thematic" environments in branches on [this repo](https://github.com/psychemedia/showntell/tree/music) but many of them may have rotted since those demos were put together.)

It would also be useful to allow a user to enter a Binderised repo URL of their own choosing.

An out there option might try to identify package imports from a static analysis of code cells in an uploaded notebook and select a Binder image that meets those requirements on the fly.

I'm not sure if there is any, let alone a widely adopted, informal convention yet for allowing a used to specify a Binder repo eg via a formulaic comment in a code cell, but this might be something to try to encourage (useful also for Jupyter Book?)

**Comment by yuvipanda** [comment]
Yeah, I think there should be UI that can attack this in a couple of different ways:

1. Select one of a list of pre-defined environments. We curate these environments and pick some
2. Enter a git repo yourself
3. Static analysis. I remember some packages that helped with this, but can't find them anymore.

For (1) and (2), we can also support keeping this state in URL query parameters, so someone can set that once and share it with users who don't need to fully understand it.

**Comment by psychemedia** [comment]
Re: static analysis, I though I had a list of tools s/where but can't find it offhand... [This](https://github.com/andrewp-as-is/list-imports.py/blob/master/list_imports/__init__.py) seems to offer a relatively simple parser?

This has a collection of possible useful stuff, eg if you wanted to feature creep and offer notebook quality metrics): https://github.com/Isabel-Gan/quantifying-notebook-features

My stalled (what a surprise!) attempt at notebook quality metrics is here: https://github.com/innovationOUtside/nb_quality_profile
